src/main/java/de/tum/in/www1/artemis/web/rest/plagiarism/PlagiarismCaseResource.java

```diff
@@ -147,6 +175,31 @@ public ResponseEntity<PlagiarismCaseInfoDTO> getPlagiarismCaseForExerciseForStud
         return ResponseEntity.ok(null);
     }
 
+    /**
+     * Retrieves plagiarismCases related to given exercise ids for the student for the exercises where the plagiarism comparison was confirmed and the student was notified
+     *
+     * @param courseId the id of the course
+     * @param exerciseIds the ids of the exercises
+     * @return a list of plagiarism case id and verdict values for the exercises only for the exercises where the comparison was confirmed and the student was notified
+     */
+    @GetMapping("courses/{courseId}/plagiarism-cases")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Map<Long, PlagiarismCaseInfoDTO>> getPlagiarismCasesForExercisesForStudent(@PathVariable long courseId,
+            @RequestParam(name = "exerciseId") Set<Long> exerciseIds) {
+        log.debug("REST request to all plagiarism cases for student and exercise with id: {}", exerciseIds);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        authenticationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, user);
+
+        var plagiarismCases = plagiarismCaseRepository.findByStudentIdAndExerciseIdsWithPost(user.getId(), exerciseIds);
+        var plagiarismCaseInfoDTOs = plagiarismCases.stream()
+                // the following line is already checked in the SQL statement, but we want to ensure it 100%
+                .filter(plagiarismCase -> plagiarismCase.getPost() != null).collect(Collectors.toMap(plagiarismCase -> plagiarismCase.getExercise().getId(),
```

Comment:
Can there be multiple cases per exercise here? In that case in the mapping of `exerciseId -> case` would contain only the one whichever came last in the query response.

Reply:
No multiple cases per exercise with the same student id is not expected here. We can confirm that by looking at the corresponding singular method [findByStudentIdAndExerciseIdWithPost](https://github.com/ls1intum/Artemis/blob/feature/plagiarism-check-for-exams/src/main/java/de/tum/in/www1/artemis/repository/plagiarism/PlagiarismCaseRepository.java#L48-L55) returns an `Optional<PlagiarismCase>`.

Reply:
Okay, in that case the name `plagiarismCases` is a bit misleading. It may also help to explicitly specify the type here instead of `var`.

Reply:
I renamed the variable and used explicit types there.

In the case you described, there is still 1 `(exerciseId, studentId)` pair created but 2 `PlagiarismSubmission` instances are linked to that. It looks like this in the instructor detail view (student view is also similar).

![image](https://user-images.githubusercontent.com/18427209/186403352-d9a320bf-f444-4b91-b759-51ac49219629.png)

![image](https://user-images.githubusercontent.com/18427209/186403361-a993510f-7010-4300-8bf4-451f0e21a776.png)